### ISSUE TITLE
Improve slide font-size

### ISSUE THREAD

**Issue by Carreau** [jupyter/nbviewer#533]
Reported on behalf of a frequent user of slide presentation:

This is not completely an issues only for nbviewer, but some improvement could be done for hosted slides. 

The browser zoom, does not affect the slide of the font. 
Though injecting a css `body{ font-size: 200%}` works. 

Idea: provide that in the menu in nbviewer in slide mode that allow to change the font-size globally. 

That would also imply that we have a way to bring back the menu from a radom slide. 

Apparently this also affect RISE, so maybe @damianavila want to do something in it, like expose a "Zoom" API. Or or is it a Reveal option ?

**Comment by dsblank** [comment]
Thank you for noting this! I meant to post something similar. I gave a talk at JupyterDay (and was "eating our own dogfood" with the slide show controls). A few people commented that the slide fonts were *way* too small. 

I suggest making the default bigger, and also allowing sane ways to zoom/unzoom.

**Comment by damianavila** [comment]
>I suggest making the default bigger,

I agree we need a bigger default...

> and also allowing sane ways to zoom/unzoom.

This is something that reveal manage by itself... on RISE you can setup width and height, but for nbconvert this is only possible using a custom template... so, we need to document how to do it and, eventually, expose different "sizes" on nbviewer, so people can adjust to their own convenience.

**Comment by damontallen** [comment]
we need to document how to do it and, eventually, expose different "sizes"
> on nbviewer, so people can adjust to their own convenience.



Please do this.

**Comment by hzzyyy** [comment]
Please do this, I'm making a slide as well. 
The fonts are too small.

**Comment by sarmadsohaib** [comment]
is it done? the python code in the slides is just way too small for the class room. please somebody do something here.

**Comment by damianavila** [comment]
This option should be provided in nbconvert and then used in nbviewer somehow (a better default with bigger fonts is probably a good idea, but probably, we also need to provide a way - maybe with the url or notebook metadata - to customize it.

**Comment by sarmadsohaib** [comment]
yes please. do something here. find a clean and nice method to improve the font sizes for class lectures presentations. 
So far what I have done is
1) change the font size to 330 by placing the code in the cell marked as RawNBConvert <style type="text/css">
.reveal {
font-size: 320%;
}
</style>
2) Change the widht and height of the html generated  by nbconvert to 2200 and 1100 respectively.

**Comment by yanivm** [comment]
For anyone interested in a temporary fix to both issues (nicer initial font size + browser zoom functionality to affect the slides): You can hack the CSS of the `.slides` element in the page by adding the following code (=essentially one line of JS) to some(*) Markdown cell in your Slideshow notebook:

    <script>
        console.log("Hello. You'll see this printed in your browser's DevTools / Console. Feel free to delete this line.");
        document.querySelector('head').innerHTML += '<style>.slides { zoom: 1.0 !important; }</style>';
    </script>

(*)any cell which is not of type 'Skip'; e.g. add it to some Markdown cell that you already intended to display in your Slideshow.

**Comment by johann-petrak** [comment]
Is there any way to influence the font used for the slide and the size of the slide within the browser window from within the notebook source code (i.e. without hacking the HTML file that gets created)? 
The suggestion by @yanivm  did not work for me.

The slide occupies only a small part of my browser window (same in Firefox and Chrome) and there are huge empty margins, especially on the left, around the slide which is really bad for presenting via sharing the screen or window.  

It is also not possible to increase the font size by the standard browser zoom keys. 

The only slight improvement so far can be achieved by editing the HTML file and adding `minScale: 1.5` to the Reveal.initialize config.

**Comment by johann-petrak** [comment]
I just discovered that things work much better when installing RISIS which allows to show the slides from within the notebook, but also handles font size, slide size within the window much better, the browser zoom key works to increase or decrease font size and even speaker notes work (which do not work with the slides created by nbconvert).
